aws/resource_aws_msk_scram_secret_test.go

```diff
@@ -0,0 +1,359 @@
+package aws
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+var newSecrets = aws.StringSlice([]string{"AmazonMSK_test_aws_example_test_tf-test-2-jbuaA1",
+	"AmazonMSK_test_aws_example_test_tf-test-5-I50lRm"})
+
+var existingSecrets = aws.StringSlice([]string{"AmazonMSK_test_aws_example_test_tf-test-2-jbuaA1",
+	"AmazonMSK_test_aws_example_test_tf-test-4-I50lRm"})
+
+func TestAccAwsMskScramSecret_basic(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_msk_sasl_scram_secret.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSMsk(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMskConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMskSaslScramSecret_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "arn"),
+					resource.TestCheckResourceAttrSet(resourceName, "scram_secrets.#"),
+				),
+			},
+		},
+	})
+}
+func TestAccAwsMskScramSecret_Delete(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_msk_sasl_scram_secret.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSMsk(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMskConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccMskSaslScramSecret_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(resourceName, "scram_secrets.#"),
+				),
+			},
+			{
+				Config: testAccMskSaslScramSecretDelete(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsSaslScramSecretsDestruction(resourceName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsMskScramSecret_UpdateRemove(t *testing.T) {
```

Comment:
Please put all the tests at the top, and the support functions below. That way it is easier to see what see what tests are included at a glance.